### ISSUE TITLE
[java] LoD false positive

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#270]
Please, prefix the report title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*

**Rule Set:** java-coupling/LawOfDemeter

**Description:**

LoD reports a false positive when method calls are placed between the var declaration and assignment.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public void example() {
        String something;
        C c = new C();
        c.doIt();
        something = "no worries";
    }
}

```

**Running PMD through:** *Any*

Placeholder report for PR #266
